File: Nov2025/src/main/java/com/nykaa/pages/ProductPage.java
package com.nykaa.pages;

import org.openqa.selenium.*;
import java.io.IOException;
import java.util.Set;
import com.nykaa.utilities.ActionHelper;
import com.nykaa.utilities.Screenshots;

public class ProductPage {
    WebDriver driver;
    ActionHelper helper;

    By firstProduct = By.cssSelector(".css-xrzmfa");
    By addToBagButton = By.cssSelector(".btn-text");

    public ProductPage(WebDriver driver) {
        this.driver = driver;
        this.helper = new ActionHelper(driver);
    }

    public void selectFirstProduct() throws IOException {
        helper.click(firstProduct, "SelectFirstProduct");

        // Switch to new product tab
        Set<String> allTabs = driver.getWindowHandles();
        for (String tab : allTabs) {
            driver.switchTo().window(tab);
        }
        System.out.println(" Switched to product tab");
        Screenshots.capture(driver, "SwitchedToProductTab");
    }

    public void addToBag() throws IOException {
        helper.click(addToBagButton, "ClickAddToBag");
        System.out.println(" Added product to bag");
    }
}

File: Pycharm_Python-main/Pycharm_Python-main/NykaaPythonNew/pages/product_page.py
import time
from selenium.webdriver.common.by import By


class ProductPage:
    def __init__(self, driver, helper):
        self.driver = driver
        self.helper = helper
        self.add_to_bag_button = (By.CSS_SELECTOR, ".btn-text")
        self.wishlist_button = (By.CSS_SELECTOR, ".wishlist_button_text.css-17jex54")
        self.product_selector = (By.CSS_SELECTOR, ".css-xrzmfa")

    def add_to_bag(self):
        self.helper.click(self.add_to_bag_button, "Click Add To Bag")
        self.helper.take_screenshot("ProductAddedToBag")


